test_runner/regress/test_replication_start_subxid_overflow.py

```diff
@@ -0,0 +1,217 @@
+import threading
+import time
+
+from fixtures.neon_fixtures import NeonEnv, wait_for_last_flush_lsn, wait_replica_caughtup
+
+
+def initdb(cur):
+    cur.execute("create table t(pk serial primary key, payload integer)")
+    cur.execute(
+        """create or replace function create_subxacts(n integer) returns void as $$
+              declare
+                 i integer;
+              begin
+                 for i in 1..n loop
+                    begin
+                       insert into t (payload) values (0);
+                    exception
+                       when others then
+                          raise exception 'caught something';
+                    end;
+                 end loop;
+              end; $$ language plpgsql"""
+    )
+
+
+def create_subxacts(env, endpoint, n_subxacts, barrier):
+    con = endpoint.connect()
+    cur = con.cursor()
+    cur.execute("create schema background")
+    cur.execute("set search_path=background")
+    initdb(cur)
+    cur.execute("begin")
+    cur.execute(f"select create_subxacts({n_subxacts})")
+    wait_for_last_flush_lsn(env, endpoint, env.initial_tenant, env.initial_timeline)
+    barrier.wait()
+    time.sleep(5)  # wait for replica start
+    cur.execute("commit")
+
+
+# This test replica startup in case of large number of active subtransactions at primary node.
+# Number of known transaction xids is limited and if we do not limit number of transactions
+# restored from CLOG, then it will cause crash of replica.
+def test_replication_start_subxid_overflow(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+
+    primary = env.endpoints.create_start(branch_name="main", endpoint_id="primary")
+
+    barrier = threading.Barrier(2)
+    thread = threading.Thread(target=create_subxacts, args=(env, primary, 100000, barrier))
+    thread.start()
+
+    # Wait until subxacts are produced at primary
+    barrier.wait()
+    secondary = env.endpoints.new_replica_start(origin=primary, endpoint_id="secondary")
+
+    # wait completion of transaction at primary
+    thread.join()
+
+    wait_replica_caughtup(primary, secondary)
+
+    con = secondary.connect()
+    cur = con.cursor()
+
+    cur.execute("set search_path = background")
+    cur.execute("select * from t where pk = 1")
+    assert cur.fetchone() == (1, 0)
+
+
+# Test starting a replica at a point in time where there is a transaction
+# running in primary with lots of subtransactions. Test MVCC in standby before
+# and after the commit of the transaction.
+#
+def test_replication_start_subxid_overflow2(neon_simple_env: NeonEnv):
```

Comment:
This test also passes without the code changes.

Reply:
I run test_replication_start_subxids_overflow at master and get the following results:
```
================================================================= 6 failed, 1 warning, 6 errors in 1601.02s (0:26:41) ==================================================================
```